airbyte-integrations/connectors/destination-pinecone/destination_pinecone/indexer.py

```diff
@@ -54,12 +54,14 @@ def delete_vectors(self, filter):
     def delete_by_metadata(self, filter, top_k):
         zero_vector = [0.0] * self.embedding_dimensions
         query_result = self.pinecone_index.query(vector=zero_vector, filter=filter, top_k=top_k)
-        vector_ids = [doc.id for doc in query_result.matches]
-        if len(vector_ids) > 0:
-            # split into chunks of 1000 ids to avoid id limit
-            batches = create_chunks(vector_ids, batch_size=MAX_IDS_PER_DELETE)
-            for batch in batches:
-                self.pinecone_index.delete(ids=list(batch))
+        while len(query_result.matches) > 0:
+            vector_ids = [doc.id for doc in query_result.matches]
+            if len(vector_ids) > 0:
+                # split into chunks of 1000 ids to avoid id limit
+                batches = create_chunks(vector_ids, batch_size=MAX_IDS_PER_DELETE)
+                for batch in batches:
+                    self.pinecone_index.delete(ids=list(batch))
```

Comment:
Just for my info, are we expecting to have to run through multiple queries? I see there's a loop through batches that should mitigate the 1000 id limit - but in that case, I'm not sure if there's also a limit applied to the query.